code/datums/components/tippable.dm

```diff
@@ -209,3 +223,19 @@
 	else
 		tipped_mob.transform = turn(tipped_mob.transform, -180)
 		REMOVE_TRAIT(tipped_mob, TRAIT_IMMOBILIZED, TIPPED_OVER)
+
+/datum/component/tippable/proc/accept_roleplay(mob/living/user, datum/emote/emote)
+	SIGNAL_HANDLER
+
+	if (!is_tipped)
+		return
+	if (roleplayed)
+		return
+	if (!is_type_in_list(emote, roleplay_emotes))
+		return
+	var/time_left = timeleft(self_untip_timer)
+	deltimer(self_untip_timer)
+	self_untip_timer = addtimer(CALLBACK(src, .proc/right_self, user), time_left / 2, TIMER_UNIQUE | TIMER_STOPPABLE)
+	roleplayed = TRUE
+
+	to_chat(user, span_notice("Your frustration has empowered you! You can now right yourself faster!"))
```

Comment:
if we're doing this, this needs to be a callback that is invoked from this component and handled on the borg itself. this is too literal

Reply:
the entire proc does?